src/compiler/transformers/ts.ts

```diff
@@ -1163,12 +1185,27 @@ namespace ts {
          * @param member The class element node.
          * @param isStatic A value indicating whether the member should be a static or instance member.
          */
-        function isInitializedProperty(member: ClassElement, isStatic: boolean) {
+        function isInitializedProperty(member: ClassElement, isStatic: boolean): member is PropertyDeclaration {

```

Comment:
This type predicate is somewhat inaccurate, since it digs further into a `PropertyDeclaration`. This results in narrowing at the call site such that `PropertyDeclaration` is possibly removed from the list of constituents of `member` even though it still could contain other `PropertyDeclaration` nodes.